### DISCUSSION TITLE
deploying contract

### DISCUSSION THREAD

**Discussion by prakharmishra3956** [Cyfrin/foundry-full-course-cu#2635] [Q&A]
![Screenshot (41)](https://github.com/user-attachments/assets/4e98d078-0224-4920-be49-0a2561d21c36)

**Comment by EngrPips** [comment]
Hello @prakharmishra3956, run `source .env`, then run `echo $ETHEREUM_RPC_URL`. Let's see what the outcome will be.

**Reply by prakharmishra3956** [reply]
![Screenshot (42)](https://github.com/user-attachments/assets/71f1b003-64bc-4087-ad3b-7909ab7a300e)

Nothing Happened after doing this.

**Reply by EngrPips** [reply]
why is there a space in between `PRIVATE_KEY`, `=` and it's value?

**Reply by EngrPips** [reply]
Also, I want to remind you not to get into the habit of showing your `.env` file publicly like this. I know the private-key here is fake but still be conscious, please.

**Reply by prakharmishra3956** [reply]
I have cleared the space between PRIVATE_KEY , = and its value. I apologize for the .env file, i will be conscious from now on.
 But nothing happened after clearing the space.

**Reply by EngrPips** [reply]
Honestly, I can't figure out what is wrong, but read this discussion https://github.com/Cyfrin/foundry-full-course-cu/discussions/2299 it might help you.

**Reply by prakharmishra3956** [reply]
I have tried all the things answered in the discussion but still there is no progress.

**Reply by EngrPips** [reply]
Arrgh, I trust someone that knows what could be wrong will see this soon enough and help. In the meantime, you can paste the information directly to the command line instead of using `.env` since that is not working for you right now.

**Comment by Shiva-Sai-ssb** [comment]
@prakharmishra3956
Is this issue resolved? I see that you didn't save the `.env` file, which might be the reason. 
Make sure to save the file after making any changes so it won't cause issues later.

**Reply by prakharmishra3956** [reply]
I tried but still nothing happened.

**Reply by Shiva-Sai-ssb** [reply]
Can you show what you get by running `echo $RPC_URL` and `echo $ETHEREUM_RPC_URL`? 
Don't forget to run `source .env` before executing those commands.

**Comment by EngrPips** [comment]
Interesting
